### PR TITLE
mappers and show data in card views

### PR DESCRIPTION
**Pull request by benjaglobant**
In this PR, I implemented the mappers to map the response into entities for show the weathers in the card views correctly. 
In the following GIF, you can see, six weathers showed in CardViews corresponding to today and the next 5 five days.
progressBar

### CODE REVIEW

**Review by BQuinteros — APPROVED** [review]

**Review by SartiMau — APPROVED** [review]

File: app/src/main/java/com/globant/weatherapp/mvp/view/WeatherView.kt [hunk 1]
```diff
@@ -1,27 +1,34 @@
...
-            layoutManager = LinearLayoutManager(this.context)
-            weatherAdapter = WeatherRecyclerViewAdapter()
             adapter = weatherAdapter
-            weatherAdapter.submitList(weathers)
+            visibility = View.VISIBLE
         }
     }
+
+    private fun initRecyclerView() {
+        activity?.recycler_view?.layoutManager = LinearLayoutManager(this.context)
```
Comment (SartiMau) [inline app/src/main/java/com/globant/weatherapp/mvp/view/WeatherView.kt]: as this method is called only from the init view, remove this method and put this line in the initView method

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/globant/weatherapp/mvp/view/WeatherView.kt (hunk 1 above)
Comment (benjaglobant) [reply app/src/main/java/com/globant/weatherapp/mvp/view/WeatherView.kt]: Done!

File: app/src/main/java/com/globant/weatherapp/data/mapper/WeatherMapper.kt [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    private fun transformWeatherByDayList(listOfWeatherByDayResponse: List<WeatherByDayResponse>): List<WeatherByDay> {
+        return listOfWeatherByDayResponse.map {
+            transformToWeatherByDay(it)
+        }
+    }
+
+    private fun transformToCity(cityResponse: CityResponse): City =
+        cityResponse.let {
+            City(
+                it.id,
```
Comment (mauroLonghi) [inline app/src/main/java/com/globant/weatherapp/data/mapper/WeatherMapper.kt]: plse, check as Harry done the transformed  fucntion, He use anohter important scope function.

Comment (benjaglobant) [reply app/src/main/java/com/globant/weatherapp/data/mapper/WeatherMapper.kt]: Done!

**Review by mauroLonghi — APPROVED** [review]

**Review by hernanhorquin96 — APPROVED** [review]

File: app/src/main/java/com/globant/weatherapp/data/mapper/WeatherMapper.kt [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+                this.id,
+                this.name
+            )
+        }
+    }
+
+    private fun transformToTemperature(temperatureResponse: TemperatureResponse): Temperature {
+        temperatureResponse.apply {
+            return Temperature(
+                this.temp,
```
Comment (tomas-kovacs) [inline app/src/main/java/com/globant/weatherapp/data/mapper/WeatherMapper.kt:54]: dont need to use "this." inside apply

Comment (benjaglobant) [reply app/src/main/java/com/globant/weatherapp/data/mapper/WeatherMapper.kt:54]: Done! Thanks!

**Review by lucianoeli — APPROVED** [review]

**Review by SartiMau — APPROVED** [review]

**Pull request merged**